platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/datasource/VirtualLeafBytes.java

```diff
@@ -237,28 +239,45 @@ public void writeTo(final WritableSequentialData out) {
     }
 
     public int getSizeInBytesForHashing() {
-        int len = 0;
-        len += 1; // 0x00 prefix
         final Bytes kb = keyBytes();
-        len += ProtoWriterTools.sizeOfDelimited(FIELD_LEAFRECORD_KEY, Math.toIntExact(kb.length()));
+        final int keyLen = Math.toIntExact(kb.length());
+        int innerLen = ProtoWriterTools.sizeOfDelimited(FIELD_LEAFRECORD_KEY, keyLen);
+
         final Bytes vb = valueBytes();
-        if (vb != null) {
-            len += ProtoWriterTools.sizeOfDelimited(FIELD_LEAFRECORD_VALUE, Math.toIntExact(vb.length()));
+        final int valueLen = (vb != null) ? Math.toIntExact(vb.length()) : -1;
+        if (valueLen >= 0) {
+            innerLen += ProtoWriterTools.sizeOfDelimited(FIELD_LEAFRECORD_VALUE, valueLen);
         }
-        return len;
+
+        // `1 +` for 0x00 prefix
+        return 1 + ProtoWriterTools.sizeOfDelimited(FIELD_MERKLELEAF_STATEITEM, innerLen);
     }
 
     // Output size must be at least getSizeInBytesForHashing()
     public void writeToForHashing(final BufferedData out) {
         out.reset();
         assert out.remaining() >= getSizeInBytesForHashing();
+
```

Comment:
Be nice to add comment for why we are adding the 0x00 byte prefix. This is because we are adding the 0x00 prefix byte to all leaf hashes in the Hiero Merkle tree. So that there is clear guaranteed domain separation of hash space between leaves and internal nodes.